From 20e668c774ce788078bf733dbdd609a8a83c7265 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Thu, 19 Jul 2018 14:07:14 +0300
Subject: [PATCH 4172/5242] MLK-18826: drm: imx: dcss: fix hang when dumping
 registers

commit  10f12d7f131e813b43178adc5efccdd49dc016ee from
https://source.codeaurora.org/external/imx/linux-imx.git

If DCSS is suspended, the clocks are disabled. Dumping the registers,
with the following command:

cat /sys/kernel/debug/imx-dcss/dump_registers

will hang the system, because of DTG.

This patch makes sure clocks are enabled before dumping the registers
and will immediately release them afterwards.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-common.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index 4b3a667..2c05d9a 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -448,6 +448,10 @@ static int dcss_trace_dump_open(struct inode *inode, struct file *file)
 
 static int dcss_dump_regs_show(struct seq_file *s, void *data)
 {
+	struct dcss_soc *dcss = s->private;
+
+	pm_runtime_get_sync(dcss->dev);
+
 	dcss_blkctl_dump_regs(s, s->private);
 	dcss_dtrc_dump_regs(s, s->private);
 	dcss_dpr_dump_regs(s, s->private);
@@ -459,6 +463,8 @@ static int dcss_dump_regs_show(struct seq_file *s, void *data)
 	dcss_hdr10_dump_regs(s, s->private);
 	dcss_ctxld_dump_regs(s, s->private);
 
+	pm_runtime_put_sync(dcss->dev);
+
 	return 0;
 }
 
-- 
1.7.9.5

